lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -134,6 +134,28 @@ def create(path)
       @zk.create(path, ignore: :node_exists)
     end
 
+    # get with retry to reduce failure when network connectivity is flaky or ZK is overloaded
+    def get(path, opts = {})
+      retry_limit = opts[:retry_limit] || 3
+      opts.delete(:retry_limit)
```

Comment:
NIT: though `opts[:retry_limit] is nil` won't throw exception when `delete`, it might not worth to do this check two times. You could simply do

```
retry_limit = 3
if opts[:retry_limit]
  retry_limit = opts[:retry_limit]
  opts.delete(:retry_limit)
end
```

Reply:
sounds good!